js/ui.js

```diff
@@ -115,13 +115,60 @@ function showRoom(player) {
 INVENTORY SCREEN FUNCTIONS -
 --------------------------*/
 
+function equipItemOnClick(player) {
+  $(".grid-item").click(function() {
+    var itemName = $(this).children("p").text();
+    for(i = 0; i < player.items.length; i++) {
+      if(player.items[i].name === itemName) {
+        if(player.items[i].type === "Weapon") {
+          player.unEquipItem(player.equippedWeapon);
+        }
+        if(player.items[i].type === "Armor") {
+          player.unEquipItem(player.equippedArmor);
+        }
+        player.equipItem(player.items[i]);
+        displayEquippedItems(player);
+      }
+    }
+  });
+}
+
+function clearInventory() {
+  $("#equipped-weapon-text").text("");
+  $("#equipped-armor-text").text("");
+
+  for(i = 0; i < player.items.length; i++) {
```

Comment:
You could probably get rid of this loop if you add a class to all slots. Then you can just reference the class with jquery and call .empty();

Reply:
the slots all have a grid-item class. the loop is needed to initialize the tooltip attributes for each inventory item and appending them to each item using jquery. i don't see any way around that. do you have an example?

Reply:
This loop on line 140 is merely emptying out every div with `#slot` id attribute right? I think you can do the same thing by just assigning a class to those divs (or referencing an existing class) and calling `.empty()` with jquery.

`$(".some-class").empty();`